src/main/java/com/google/cloud/pontem/AvroUtil.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.pontem;
+
+import com.google.cloud.spanner.Type;
+import java.util.List;
+import java.util.logging.Logger;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaBuilder;
+
+/** Utils for Avro */
+public class AvroUtil {
+  private static final Logger LOG = Logger.getLogger(AvroUtil.class.getName());
+  public static final String AVRO_SCHEMA_FOLDER_LOCATION = "avro_schemas";
+  public static final String AVRO_SCHEMA_NAMESPACE = "com.google.cloud.pontem";
+  public static final String AVRO_RECORD_VERSION = "1";
+
+  /**
+   * Since the Avro column schema could be a UNION of types (e.g., [STRING, NULL]), we need to
+   * unpack that UNION to simply be a noramlly recognized Avro type (e.g., STRING).
+   */
+  public static Schema getSingleAvroTypeFromNullableUnion(Schema columnSchema) {
+    if (columnSchema.getType() == Schema.Type.UNION) {
+      List<Schema> columnUnionAvroTypes = columnSchema.getTypes();
+      if (columnUnionAvroTypes.size() != 2) {
+        throw new IllegalArgumentException("AvroRecord contains an unexpected format");
+      } else if (columnUnionAvroTypes.get(0).getType() == Schema.Type.NULL) {
+        return columnUnionAvroTypes.get(1);
+      } else if (columnUnionAvroTypes.get(1).getType() == Schema.Type.NULL) {
+        return columnUnionAvroTypes.get(0);
+      } else {
+        throw new IllegalArgumentException("AvroRecord contains an unexpected format");
+      }
+    }
+    throw new IllegalArgumentException("AvroRecord contains an unexpected format");
```

Comment:
Is this secocnd throw intended? My understanding from the comment is that the column could be a union but this is enforcing that it is one. Looking at the calling code you do this because you are already checking that this is a union before calling this but, wouldn't this method be more useful if you just return the schema type regardless of whether the schema is a union or not? therefore unburdening the client code from knowing how to handle different Avro schema formats?

Reply:
IMHO it is better for getSingleAvroTypeFromNullableUnion to throw the second exception because getSingleAvroTypeFromNullableUnion should only be called on a union. If it is not called on a union and we silently keep going, we risk a problem down the road.